File: cxx/src/Creational/Prototype/AbstractFork.cpp
0


//AbstractFork.java - Two of Two Prototypes

public abstract class AbstractFork implements Cloneable
{  
   String forkName; 
    
   public void setForkName(String forkName) {
       this.forkName = forkName;
   }
   public String getForkName() {
       return this.forkName;
   }
   
   public Object clone() 
   {
       Object object = null;
       try {
           object = super.clone();
       } catch (CloneNotSupportedException exception) {
           System.err.println("AbstractFork is not Cloneable");
       }
       return object;
   }   
}

File: javascript/src/Creational/Prototype/AbstractFork.js
(function() {
  define([], function() {
    var AbstractFork;
    AbstractFork = (function() {
      function AbstractFork() {
        throw "Can't instantiate abstract class, AbstractFork";
      }

      AbstractFork.prototype.forkName = null;

      AbstractFork.prototype.setForkName = function(forkName) {
        return this.forkName = forkName;
      };

      AbstractFork.prototype.getForkName = function() {
        return this.forkName;
      };

      AbstractFork.prototype.clone = function() {
        return Object.create(this);
      };

      return AbstractFork;

    })();
    return AbstractFork;
  });

}).call(this);
